### PR TITLE
[FEAT] Treasure Island UI

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

This PR add the UI for treasure island. This includes diggable sand plots and all visuals to go with this:
- Goblin Digging
- Goblin Emotion (Happy or Bleh depending on what was found)
- Updated shop modal (mobile)
- No Sand Shovel interaction/warnings

**I have a small BE PR that is needed to have the dug coordinates persisted**

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Go to Treasure Island and dig for treasure
- Only one plot should be able to be dug at a time

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by adamhannigan** [comment]
Nice work mate @spencerdezartsmith . Some non-blocking UI suggestions:

- The unhappy face is hidden behind the NPC digging goblin. This means that nothing is shown an an empty dig - it felt a bit confusing.
- It took me awhile to realise to click on the resource again to claim it. Could we add a shake effect after a few seconds if they don't claim it?
- Consider using the `img-highlight-heavy` on the image, the white border was not that bright against the sand.
- When the modal pops up when claiming a award, the hole that was dig briefly disappears. This causes a flash effect.
- It still feels like we are missing some feedback when hovering over the land to indicate it is clickable and what specific grid space they are clicking on. Is it possible to show a show or something to display which square will get dug up?

**Comment by adamhannigan** [comment]
We will want to exclude `src/assets/npcs/goblin_treasure_sheet.png` from this repo - consider adding into the private repo

File: src/features/treasureIsland/components/SandPlot.tsx [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+              the southern end of the island.
+            </p>
+          </div>
+          <Button onClick={acknowledgeNoSandShovel}>Got it</Button>
+        </CloseButtonPanel>
+      </Modal>
+    );
+  }
+
+  const isDigging =
```
Comment (adamhannigan) [inline src/features/treasureIsland/components/SandPlot.tsx:267]: Why can't we use `sandPlotState.matches("digging")`? it sounds like that state indicates the Goblin must be digging

File: src/features/treasureIsland/lib/sandPlotMachine.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+export type SandPlotState = {
+  value:
+    | "loading"
+    | "idle"
+    | "digging"
+    | "noShovel"
+    | "treasureFound"
+    | "treasureNotFound"
+    | "opacityTransition"
```
Comment (adamhannigan) [inline src/features/treasureIsland/lib/sandPlotMachine.ts:18]: Consider naming this something that doesn't leak the implementation details.

`disappearing` or `finishing`

**Review by adamhannigan — APPROVED** [review]
LGTM. Just a few small code suggestions.

There is a non API error that occassional arises but I'm having difficulty reproducing it

File: src/features/treasureIsland/components/SandPlot.tsx (hunk 1 above)
Comment (spencerdezartsmith) [reply src/features/treasureIsland/components/SandPlot.tsx:267]: We need to be able to see the goblin AFTER the action of digging is complete. Its the time when he has finished digging but pausing for the reveal of the reward.

**Pull request merged**